Sources/Basic/OutputByteStream.swift

```diff
@@ -521,3 +523,73 @@ public final class BufferedOutputByteStream: OutputByteStream {
         contents += bytes
     }
 }
+
+/// Represents a stream which is backed to a file. Not for instantiating.
+public class FileOutputByteStream: OutputByteStream {
+
+    /// Closes the file flushing any buffered data.
+    public func close() throws {
+        fatalError("close() should be implemented by a subclass")
+    }
+}
+
+/// Implements file output stream for local file system.
+public final class LocalFileOutputByteStream: FileOutputByteStream {
+
+    /// The pointer to the file.
+    let fp: UnsafeMutablePointer<FILE>
+
+    /// True if there were any IO error during writing.
+    private var error: Bool = false
+
+    /// Instantiate using the file pointer.
+    init(fp: UnsafeMutablePointer<FILE>) throws {
+        self.fp = fp
+        super.init()
+    }
+
+    /// Opens the file for writing at the provided path.
+    public init(_ path: AbsolutePath) throws {
+        guard let fp = fopen(path.asString, "wb") else {
+            throw FileSystemError(errno: errno)
+        }
+        self.fp = fp
+        super.init()
+    }
+
+    func errorDetected() {
+        error = true
+    }
+
+    override final func writeImpl<C: Collection where C.Iterator.Element == UInt8>(_ bytes: C) {
+        // FIXME: This might be copying bytes.
+        var contents = [UInt8](bytes)
+        while true {
+            let n = fwrite(&contents, 1, contents.count, fp)
+            if n < 0 {
+                if errno == EINTR { continue }
+                errorDetected()
+            } else if n != contents.count {
+                errorDetected()
+            }
+            break
+        }
+        // Flush to file.
+        fflush(fp)
+    }
+
+    override public func close() throws {
+        defer { fclose(fp) }
+        flush()
+        // Throw if errors were found during writing.
+        if error {
+            throw FileSystemError.ioError
+        }
+    }
+}
+
+/// Public stdout stream instance.
+public var stdoutStream: FileOutputByteStream = try! LocalFileOutputByteStream(fp: libc.stdout)
+
+/// Public stderr stream instance.
+public var stderrStream: FileOutputByteStream = try! LocalFileOutputByteStream(fp: libc.stderr)
```

Comment:
+1 to both of these.

I don't think they should close the stream on de-init though, can we get a special initializer for that? The case I am worried about is where during program teardown something closes stdout/stderr early and then other things running as part of shutdown can't log errors.